src/hashfunctions.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Datastructures: GAP package providing common datastructures.
+ * Licensed under the GPL 2 or later.
+ *
+ * hashfun: various hash functions
+ */
+
+#ifndef HASHFUNCTIONS_H
+#define HASHFUNCTIONS_H
+
+#include "datastructures.h"
+#include "src/intfuncs.h"
+
+extern struct DatastructuresModule HashFunctionsModule;
+
+
+// Hash two integers together
+static inline UInt HashCombine2(UInt hash1, UInt hash2);
+
+// Transform a UInt into a signed GAP intermediate integer, shrinking
+// the size of the number as required
+static inline Obj HashValueToObjInt(UInt uhash)
+{
+    Int hash = (Int)uhash;
+    // Make sure bottom bits are not lost
+    hash += hash << 11;
+    hash /= 16;
+    return INTOBJ_INT(hash);
+}
+
+// Perform a shuffle of a UInt. Ideally, changing any
+// bit would have a 50/50 chance of changing every other bit.
+// The main purpose of this is to allow adding values when
+// we want to hash an unordered list -- while 1+3 = 2+2,
+// HashUInt(1) + HashUInt(3) should not be equal to
+// HashUInt(2) + HashUInt(2)
+// From: http://www.concentric.net/~Ttwang/tech/inthash.htm
+static UInt ShuffleUInt(UInt key);
```

Comment:
Why are this comment and declaration here? Wouldn't it make more semse to put the implementation here, remov the decl,, and then follow everything by HashCombine2/3?